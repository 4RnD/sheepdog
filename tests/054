#!/bin/bash

# Test stale objects purging with different disk size.
seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup
#start three in different size
for i in 0 1 2; do
	_start_sheep $i "-s $(($i+1))000"
done
_wait_for_sheep 3
$COLLIE cluster format
sleep 1

$COLLIE vdi create test 100M -P
$COLLIE cluster info | _filter_cluster_info

#start recovery
 _start_sheep 3 "-s 3000"

_wait_for_sheep 4
_wait_for_sheep_recovery 0
$COLLIE cluster info | _filter_cluster_info

#test no object in .stale
ls $STORE/*/*/.stale | grep ^- | wc -l
