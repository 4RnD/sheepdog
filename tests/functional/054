#!/bin/bash

# Test stale objects purging with different disk size.
seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_need_to_be_root

_cleanup

_make_device 0 $((1024 ** 3))
_make_device 1 $((2 * 1024 ** 3))
_make_device 2 $((3 * 1024 ** 3))
_make_device 3 $((3 * 1024 ** 3))

#start three in different size
for i in 0 1 2; do
	_start_sheep $i
done
_wait_for_sheep 3
$COLLIE cluster format
sleep 1

$COLLIE vdi create test 100M -P
$COLLIE cluster info | _filter_cluster_info

#start recovery
_start_sheep 3

_wait_for_sheep 4
_wait_for_sheep_recovery 0
$COLLIE cluster info | _filter_cluster_info

#test no object in .stale
ls $STORE/*/*/.stale | grep ^- | wc -l
