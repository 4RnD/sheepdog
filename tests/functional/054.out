QA output created by 054
using backend plain store
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:42, 127.0.0.1:7001:85, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:171, 127.0.0.1:7005:214]
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:37, 127.0.0.1:7001:74, 127.0.0.1:7002:112, 127.0.0.1:7003:112, 127.0.0.1:7004:149, 127.0.0.1:7005:187, 127.0.0.1:7006:224]
DATE      1 [127.0.0.1:7000:42, 127.0.0.1:7001:85, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:171, 127.0.0.1:7005:214]
0
